.pre-commit-config.yaml

```diff
@@ -61,7 +61,7 @@ repos:
       exclude: ^vendor|^docs
 
 - repo: https://github.com/golangci/golangci-lint
-  rev: v1.55.2
+  rev: v1.57.2
```

Comment:
does this compatible with golang 1.20?

Reply:
As far as i can read, they're adding compatiblity with newer golang version without dropping compatilbity with older versions.

(e.g. support for golang 1.21 has been added in version [1.54](https://golangci-lint.run/product/changelog/#v1540).

Reply:
https://golangci-lint.run/welcome/faq/#which-go-versions-are-supported

> Which Go versions are supported
The same as the Go team (the 2 latest minor versions).

@jpodivin ack, probably they need go version 1.21 but they're compatible with go 1.{20,21,22} I'll check with version 1.56.x

Reply:
based on my local run on this PR and on the CI job this version of pre-commit does not work with golang 1.20
```
❯ pre-commit run --all
[INFO] Initializing environment for https://github.com/golangci/golangci-lint.
[INFO] Installing environment for https://github.com/golangci/golangci-lint.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/usr/local/bin/go', 'install', './...')
return code: 1
stdout: (none)
stderr:
    go: downloading golang.org/x/tools v0.19.0
    go: downloading github.com/mattn/go-colorable v0.1.13
    go: downloading github.com/sirupsen/logrus v1.9.3
    go: downloading github.com/sourcegraph/go-diff v0.7.0
    go: downloading github.com/stretchr/testify v1.9.0
    go: downloading github.com/go-viper/mapstructure/v2 v2.0.0-alpha.1
    go: downloading github.com/golangci/revgrep v0.5.2
    go: downloading github.com/fatih/color v1.16.0
    go: downloading honnef.co/go/tools v0.4.7
    go: downloading github.com/spf13/viper v1.12.0
    go: downloading github.com/spf13/pflag v1.0.5
    go: downloading gopkg.in/yaml.v3 v3.0.1
    go: downloading github.com/golangci/plugin-module-register v0.1.1
    go: downloading golang.org/x/exp v0.0.0-20240103183307-be819d1f06fc
    go: downloading github.com/hashicorp/go-version v1.6.0
    go: downloading github.com/mitchellh/go-homedir v1.1.0
    go: downloading github.com/ldez/gomoddirectives v0.2.4
    go: downloading github.com/pelletier/go-toml/v2 v2.2.0
    go: downloading github.com/santhosh-tekuri/jsonschema/v5 v5.3.1
    go: downloading github.com/gofrs/flock v0.8.1
    go: downloading github.com/spf13/cobra v1.7.0
    go: downloading go.uber.org/automaxprocs v1.5.3
    go: downloading github.com/go-xmlfmt/xmlfmt v1.1.2
    go: downloading 4d63.com/gocheckcompilerdirectives v1.2.1
    go: downloading 4d63.com/gochecknoglobals v0.2.1
    go: downloading github.com/4meepo/tagalign v1.3.3
    go: downloading github.com/Abirdcfly/dupword v0.0.14
    go: downloading github.com/Antonboom/errname v0.1.12
    go: downloading github.com/Antonboom/nilnil v0.1.7
    go: downloading github.com/Antonboom/testifylint v1.2.0
    go: downloading github.com/BurntSushi/toml v1.3.2
    go: downloading github.com/Djarvur/go-err113 v0.0.0-20210108212216-aea10b59be24
    go: downloading github.com/GaijinEntertainment/go-exhaustruct/v3 v3.2.0
    go: downloading github.com/alecthomas/go-check-sumtype v0.1.4
    go: downloading github.com/OpenPeeDeeP/depguard/v2 v2.2.0
    go: downloading github.com/alexkohler/nakedret/v2 v2.0.4
    go: downloading github.com/alexkohler/prealloc v1.0.0
    go: downloading github.com/alingse/asasalint v0.0.11
    go: downloading github.com/ashanbrown/forbidigo v1.6.0
    go: downloading github.com/ashanbrown/makezero v1.1.1
    go: downloading github.com/bkielbasa/cyclop v1.2.1
    go: downloading github.com/bombsimon/wsl/v4 v4.2.1
    go: downloading github.com/blizzy78/varnamelen v0.8.0
    go: downloading github.com/breml/bidichk v0.2.7
    go: downloading github.com/butuzov/ireturn v0.3.0
    go: downloading github.com/breml/errchkjson v0.3.6
    go: downloading github.com/butuzov/mirror v1.1.0
    go: downloading github.com/pelletier/go-toml v1.9.5
    go: downloading github.com/catenacyber/perfsprint v0.7.1
    go: downloading github.com/charithe/durationcheck v0.0.10
    go: downloading github.com/ckaznocha/intrange v0.1.1
    go: downloading github.com/curioswitch/go-reassign v0.2.0
    go: downloading github.com/daixiang0/gci v0.12.3
    go: downloading github.com/denis-tingaikin/go-header v0.5.0
    go: downloading github.com/firefart/nonamedreturns v1.0.4
    go: downloading github.com/fzipp/gocyclo v0.6.0
    go: downloading github.com/ghostiam/protogetter v0.3.5
    go: downloading github.com/go-critic/go-critic v0.11.2
    go: downloading github.com/golangci/dupl v0.0.0-20180902072040-3e9179ac440a
    go: downloading github.com/golangci/gofmt v0.0.0-20231018234816-f50ced29576e
    go: downloading github.com/golangci/misspell v0.4.1
    go: downloading github.com/golangci/unconvert v0.0.0-20240309020433-c5143eacb3ed
    go: downloading github.com/gordonklaus/ineffassign v0.1.0
    go: downloading github.com/gostaticanalysis/forcetypeassert v0.1.0
    go: downloading github.com/gostaticanalysis/nilerr v0.1.1
    go: downloading github.com/hexops/gotextdiff v1.0.3
    go: downloading github.com/jgautheron/goconst v1.7.1
    go: downloading github.com/jirfag/go-printf-func-name v0.0.0-20200119135958-7558a9eaa5af
    go: downloading github.com/jingyugao/rowserrcheck v1.1.1
    go: downloading github.com/jjti/go-spancheck v0.5.3
    go: downloading github.com/julz/importas v0.1.0
    go: downloading github.com/karamaru-alpha/copyloopvar v1.0.10
    go: downloading github.com/kisielk/errcheck v1.7.0
    go: downloading github.com/kkHAIKE/contextcheck v1.1.5
    go: downloading github.com/kulti/thelper v0.6.3
    go: downloading github.com/kunwardeep/paralleltest v1.0.10
    go: downloading github.com/kyoh86/exportloopref v0.1.11
    go: downloading github.com/ldez/tagliatelle v0.5.0
    go: downloading github.com/leonklingele/grouper v1.1.1
    go: downloading github.com/lufeee/execinquery v1.2.1
    go: downloading github.com/macabu/inamedparam v0.1.3
    go: downloading github.com/maratori/testableexamples v1.0.0
    go: downloading github.com/maratori/testpackage v1.1.1
    go: downloading github.com/matoous/godox v0.0.0-20230222163458-006bad1f9d26
    go: downloading github.com/mgechev/revive v1.3.7
    go: downloading github.com/moricho/tparallel v0.3.1
    go: downloading github.com/nakabonne/nestif v0.3.1
    go: downloading github.com/nishanths/exhaustive v0.12.0
    go: downloading github.com/nishanths/predeclared v0.2.2
    go: downloading github.com/nunnatsa/ginkgolinter v0.16.2
    go: downloading github.com/polyfloyd/go-errorlint v1.4.8
    go: downloading github.com/ryancurrah/gomodguard v1.3.1
    go: downloading github.com/ryanrolds/sqlclosecheck v0.5.1
    go: downloading github.com/sanposhiho/wastedassign/v2 v2.0.7
    go: downloading github.com/sashamelentyev/interfacebloat v1.1.0
    go: downloading github.com/sashamelentyev/usestdlibvars v1.25.0
    go: downloading github.com/securego/gosec/v2 v2.19.0
    go: downloading github.com/shazow/go-diff v0.0.0-20160112020656-b6b7b6733b8c
    go: downloading github.com/sivchari/containedctx v1.0.3
    go: downloading github.com/sivchari/tenv v1.7.1
    go: downloading github.com/sonatard/noctx v0.0.2
    go: downloading github.com/ssgreg/nlreturn/v2 v2.2.1
    go: downloading github.com/stbenjam/no-sprintf-host-port v0.1.1
    go: downloading github.com/tdakkota/asciicheck v0.2.0
    go: downloading github.com/tetafro/godot v1.4.16
    go: downloading github.com/timakin/bodyclose v0.0.0-20230421092635-574207250966
    go: downloading github.com/timonwong/loggercheck v0.9.4
    go: downloading github.com/tomarrell/wrapcheck/v2 v2.8.3
    go: downloading github.com/tommy-muehle/go-mnd/v2 v2.5.1
    go: downloading github.com/ultraware/funlen v0.1.0
    go: downloading github.com/ultraware/whitespace v0.1.0
    go: downloading github.com/uudashr/gocognit v1.1.2
    go: downloading github.com/xen0n/gosmopolitan v1.2.2
    go: downloading github.com/yagipy/maintidx v1.0.0
    go: downloading github.com/yeya24/promlinter v0.2.0
    go: downloading github.com/ykadowak/zerologlint v0.1.5
    go: downloading gitlab.com/bosi/decorder v0.4.1
    go: downloading go-simpler.org/musttag v0.9.0
    go: downloading go-simpler.org/sloglint v0.5.0
    go: downloading mvdan.cc/gofumpt v0.6.0
    go: downloading mvdan.cc/unparam v0.0.0-20240104100049-c549a3470d14
    go: downloading github.com/mattn/go-isatty v0.0.20
    go: downloading golang.org/x/mod v0.16.0
    go: downloading golang.org/x/sys v0.18.0
    go: downloading github.com/davecgh/go-spew v1.1.1
    go: downloading github.com/stretchr/objx v0.5.2
    go: downloading github.com/fsnotify/fsnotify v1.5.4
    go: downloading github.com/pmezard/go-difflib v1.0.0
    go: downloading github.com/mitchellh/mapstructure v1.5.0
    go: downloading github.com/spf13/afero v1.11.0
    go: downloading github.com/spf13/cast v1.5.0
    go: downloading github.com/spf13/jwalterweatherman v1.1.0
    go: downloading github.com/gobwas/glob v0.2.3
    go: downloading golang.org/x/sync v0.6.0
    go: downloading go.uber.org/zap v1.24.0
    go: downloading github.com/fatih/structtag v1.2.0
    go: downloading github.com/gostaticanalysis/comment v1.4.2
    go: downloading golang.org/x/text v0.14.0
    go: downloading github.com/google/go-cmp v0.6.0
    go: downloading gopkg.in/yaml.v2 v2.4.0
    go: downloading github.com/gostaticanalysis/analysisutil v0.7.1
    go: downloading github.com/ettle/strcase v0.2.0
    go: downloading github.com/Masterminds/semver v1.5.0
    go: downloading github.com/ccojocar/zxcvbn-go v1.0.2
    go: downloading github.com/go-toolsmith/astcast v1.1.0
    go: downloading github.com/go-toolsmith/astcopy v1.1.0
    go: downloading github.com/go-toolsmith/astequal v1.2.0
    go: downloading github.com/go-toolsmith/astfmt v1.1.0
    go: downloading github.com/go-toolsmith/astp v1.1.0
    go: downloading github.com/go-toolsmith/strparse v1.1.0
    go: downloading github.com/go-toolsmith/typep v1.1.0
    go: downloading github.com/quasilyte/go-ruleguard v0.4.2
    go: downloading github.com/quasilyte/regex/syntax v0.0.0-20210819130434-b3f0c404a727
    go: downloading github.com/prometheus/client_golang v1.12.1
    go: downloading github.com/prometheus/client_model v0.2.0
    go: downloading golang.org/x/exp/typeparams v0.0.0-20240314144324-c7f7c6466f7f
    go: downloading github.com/subosito/gotenv v1.4.1
    go: downloading github.com/hashicorp/hcl v1.0.0
    go: downloading gopkg.in/ini.v1 v1.67.0
    go: downloading github.com/magiconair/properties v1.8.6
    go: downloading github.com/chavacava/garif v0.1.0
    go: downloading github.com/olekukonko/tablewriter v0.0.5
    go: downloading go.uber.org/multierr v1.6.0
    go: downloading go.uber.org/atomic v1.7.0
    go: downloading github.com/golang/protobuf v1.5.3
    go: downloading github.com/mattn/go-runewidth v0.0.9
    go: downloading google.golang.org/protobuf v1.33.0
    go: downloading github.com/cespare/xxhash/v2 v2.1.2
    go: downloading github.com/beorn7/perks v1.0.1
    go: downloading github.com/prometheus/procfs v0.7.3
    go: downloading github.com/prometheus/common v0.32.1
    go: downloading github.com/quasilyte/gogrep v0.5.0
    go: downloading github.com/quasilyte/stdinfo v0.0.0-20220114132959-f7386bf02567
    go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
    pkg/config/loader.go:8:2: package slices is not in GOROOT (/usr/lib/go-1.20/src/slices)
    note: imported by a module that requires go 1.21
Check the log at /home/gibi/.cache/pre-commit/pre-commit.log
```